app/api/api_v1/routers/cclw_ingest.py

```diff
@@ -0,0 +1,484 @@
+
+from app.api.api_v1.schemas.document import (
+    BulkIngestResult,
+    DocumentUpdateRequest,
+)
+from app.core.auth import get_superuser_details
+from app.core.aws import get_s3_client
+from app.core.cclw_ingestion.ingest_row_cclw import (
+    CCLWDocumentIngestRow,
+    EventIngestRow,
+)
+from app.core.cclw_ingestion.pipeline import generate_pipeline_ingest_input
+from app.core.ingestion.processor import (
+    initialise_context,
+    get_dfc_ingestor,
+    get_dfc_validator,
+    get_event_ingestor,
+)
+from app.core.cclw_ingestion.reader import get_file_contents, read
+from app.core.ingestion.utils import (
+    IngestContext,
+    Result,
+    ResultType,
+    ValidationResult,
+    get_result_counts,
+)
+from app.core.ingestion.validator import validate_event_row
+from app.core.validation import IMPORT_ID_MATCHER
+from app.core.validation.types import ImportSchemaMismatchError
+from app.core.validation.util import (
+    get_new_s3_prefix,
+    write_csv_to_s3,
+    write_documents_to_s3,
+    write_ingest_results_to_s3,
+)
+from app.db.models.document.physical_document import PhysicalDocument
+from app.db.models.law_policy.family import FamilyDocument, Slug
+from app.db.session import get_db
+
+_LOGGER = logging.getLogger(__name__)
+
+cclw_ingest_router = r = APIRouter()
+
+
+def _start_ingest(
+    db: Session,
+    s3_client: S3Client,
+    s3_prefix: str,
+    documents_file_contents: str,
+    events_file_contents: str,
+):
+    context = None
+    # TODO: add a way for a user to monitor progress of the ingest
+    try:
+        context = initialise_context(db, "CCLW")
+        document_ingestor = get_dfc_ingestor(db)
+        read(documents_file_contents, context, CCLWDocumentIngestRow, document_ingestor)
+        event_ingestor = get_event_ingestor(db)
+        read(events_file_contents, context, EventIngestRow, event_ingestor)
+    except Exception as e:
+        # This is a background task, so do not raise
+        _LOGGER.exception(
+            "Unexpected error on ingest", extra={"props": {"errors": str(e)}}
+        )
+
+    try:
+        if context is not None:
+            write_ingest_results_to_s3(
+                s3_client=s3_client,
+                s3_prefix=s3_prefix,
+                results=context.results,
+            )
+    except Exception as e:
+        _LOGGER.exception(
+            "Unexpected error writing ingest results to s3",
+            extra={"props": {"errors": str(e)}},
+        )
+
+    try:
+        pipeline_ingest_input = generate_pipeline_ingest_input(db)
+        write_documents_to_s3(
+            s3_client=s3_client,
+            s3_prefix=s3_prefix,
+            documents=pipeline_ingest_input,
+        )
+    except Exception as e:
+        _LOGGER.exception(
+            "Unexpected error writing pipeline input document to s3",
+            extra={"props": {"errors": str(e)}},
+        )
+
+
+@r.post(
+    "/bulk-ingest/validate/cclw/law-policy",
```

Comment:
shall we just remove the `law-policy` part from these URLs?

Reply:
Yeah - I have that on my list ... its last for obvious reasons :+1: